src/backend/distributed/utils/background_jobs.c

```diff
@@ -741,6 +747,53 @@ TaskEnded(TaskExecutionContext *taskExecutionContext)
 }
 
 
+/*
+ * QueueMonitorSigHupHandler handles SIGHUP to update monitor related config params.
+ */
+static void
+QueueMonitorSigHupHandler(SIGNAL_ARGS)
+{
+	GotSighup = true;
```

Comment:
The signal handlers for the normal maintenance daemon contain some code related to saving `errno`. That sounds like it could be important. Probably nice to share these handlers.

Reply:
Maybe we can really make these functions shared across the different background daemons that we have? Because they have the exact same body afaict.

Reply:
I think I should investigate that in another PR.